### PR TITLE
update contrib guide, add developer notes

### PR DESCRIPTION
**Pull request by clewis7**
supersedes #580

Because something wonky happened with the iris dataset and all of the changes with imgui, it was easier to just open a new PR for this

### CODE REVIEW

**Comment by kushalkolar** [comment]
Some formatting issues and also add the developer docs as a separate dir, like user guide, and have a toc page with the graphics and plotarea in the toctree

File: docs/source/developer_notes/graphics.rst [hunk 1]
```diff
@@ -61,10 +61,10 @@ It defines the line, and provides additional useful functionality.
 For example, every time that the ``data`` is changed, the new data will be marked for upload to the GPU before the next draw.
 In addition, event handlers will be called if any event handlers are registered.
 
-``VertexColors``behaves similarly, but it can perform additional parsing that can create the colors buffer from different
+``VertexColors`` behaves similarly, but it can perform additional parsing that can create the colors buffer from different
 forms of user input. For example if a user runs: ``line_graphic.colors = "blue"``, then ``VertexColors.__setitem__()`` will
 create a buffer that corresponds to what ``pygfx.Color`` thinks is "blue". Users can also take advantage of fancy indexing,
-ex: ``line_graphics.colors[bool_array] = "red"`` :smile:
+ex: ``line_graphics.colors[bool_array] = "red"``
```
Comment (kushalkolar) [inline docs/source/developer_notes/graphics.rst:67]: You can use the unicode emoji instead: 😄

In KDE Plasma you can get the emoji selector from the app menu

**Review by kushalkolar — CHANGES_REQUESTED** [review]

File: docs/source/developer_notes/graphics.rst [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+Now look at the constructor for the ``LineGraphic`` base class ``PositionsGraphic``, it first creates an instance of ``VertexPositions``.
+This is a class that manages vertex positions buffer. For the user, it defines the line data, and provides additional useful functionality.
+It defines the line, and provides additional useful functionality.
+For example, every time that the ``data`` is changed, the new data will be marked for upload to the GPU before the next draw.
+In addition, event handlers will be called if any event handlers are registered.
+
+``VertexColors`` behaves similarly, but it can perform additional parsing that can create the colors buffer from different
+forms of user input. For example if a user runs: ``line_graphic.colors = "blue"``, then ``VertexColors.__setitem__()`` will
+create a buffer that corresponds to what ``pygfx.Color`` thinks is "blue". Users can also take advantage of fancy indexing,
+ex: ``line_graphics.colors[bool_array] = "red"``
```
Comment (clewis7) [inline docs/source/developer_notes/graphics.rst:67]:
```suggestion
ex: ``line_graphics.colors[bool_array] = "red"`` 😊
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,221 +1,299 @@
...
+This guide explains how to contribute: if you have questions about the process, please 
+reach out on [GitHub Discussions](https://github.com/fastplotlib/fastplotlib/discussions).
+
+> **_NOTE:_** If you are already familiar with contributing to open-source software packages, 
+> please check out the [quick guide](#contributing-quick-guide)!
+
+## General Guidelines
+
+Developers are encouraged to contribute to various areas of development. This could include the addition of new features (e.g. 
+graphics or selector tools), bug fixes, or the addition of new examples to the [examples gallery](https://fastplotlib.readthedocs.io/en/latest/_gallery/index.html).
```
Comment (kushalkolar) [inline CONTRIBUTING.md:16]: update link to our new domain and /ver/dev

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,221 +1,299 @@
...
+Developers are encouraged to contribute to various areas of development. This could include the addition of new features (e.g. 
+graphics or selector tools), bug fixes, or the addition of new examples to the [examples gallery](https://fastplotlib.readthedocs.io/en/latest/_gallery/index.html). 
+Enhancements to documentation and the overall readability of the code are also greatly appreciated. 
+
+Feel free to work on any section of the code that you believe you can improve. More importantly, remember to thoroughly test all
+your classes and functions, and to provide clear, detailed comments within your code. This not only aids others in using the library,
+but also facilitates future maintenance and further development.
+
+For more detailed information about `fastplotlib` modules, including design choices and implementation details, visit the 
+[`For Develeopers`](https://fastplotlib.readthedocs.io/en/latest/developer_notes/index.html) section of the package documentation.
```
Comment (kushalkolar) [inline CONTRIBUTING.md:24]: update link to our new domain and /ver/dev

**Review by kushalkolar — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 3 above)
Comment (clewis7) [inline CONTRIBUTING.md:16]:
```suggestion
graphics or selector tools), bug fixes, or the addition of new examples to the [examples gallery](https://www.fastplotlib.org/ver/dev/). 
```

File: CONTRIBUTING.md (hunk 4 above)
Comment (clewis7) [inline CONTRIBUTING.md:24]:
```suggestion
[`For Develeopers`](https://www.fastplotlib.org/ver/dev/developer_notes/index.html) section of the package documentation.
```

**Review by kushalkolar — APPROVED** [review]

**Pull request merged**